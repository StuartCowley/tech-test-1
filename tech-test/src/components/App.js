import React, { useState } from "react"
import SearchResults from "./SearchResults";
import Search from "./Search";
import "../styles/App.css";

function App() {
  const [ searchResults, setSearchResults ] = useState([]);

  return (
    <div className="App">
      <img 
        className="nasa-logo" 
        alt="nasa-logo" 
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg">
      </img>
      <Search setSearchResults={setSearchResults} />
      <SearchResults results={searchResults} />
    </div>
  );
};

export default App;
