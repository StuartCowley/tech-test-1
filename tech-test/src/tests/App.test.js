import { render, screen } from "@testing-library/react";
import React from "react";
import App from "../components/App";

describe("App", () => {
  it("renders App component correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders NASA logo", () => {
    render(<App />);
    const nasaLogo = screen.getByAltText("nasa-logo")
    expect(nasaLogo).toHaveClass("nasa-logo");
    expect(nasaLogo).toBeInTheDocument();
  });
});
