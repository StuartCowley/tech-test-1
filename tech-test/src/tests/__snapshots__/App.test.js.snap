// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="nasa-logo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      class="Search"
    >
      <form
        class="seach-form"
      >
        <input
          class="search-input"
          placeholder="Search Images"
          type="text"
        />
        <button
          class="search-button"
          type="submit"
        >
          Launch!
        </button>
      </form>
    </div>
    <p>
      No results found
    </p>
  </div>
</DocumentFragment>
`;
